cirq/protocols/json_serialization.py

```diff
@@ -468,6 +607,32 @@ def to_json(
             party classes, prefer adding the _json_dict_ magic method
             to your classes rather than overriding this default.
     """
+    if cls == CirqEncoder and has_serializable_by_keys(obj):
```

Comment:
It will in theory, but mypy complained about using `cls` as a base type for a class. I can try a bit more wrangling here if we're interested in getting this feature for other encoders (although what other encoders do we use?)

Reply:
Well, people could extend the CirqEncoder to their liking and pass it in. At least that's what the interface is saying. Do we depend on the CirqEncoder anywhere in the SerializableKey design? I don't think so - in which case introducing business logic just to pacify mypy seems wrong - we should just add `# type: ignore` in case we can't get around it.

Reply:
Opted for `# type: ignore`. A brief glance at mypy Github issues seemed to suggest that this is a known issue.